main.cpp

```diff
@@ -45,8 +46,43 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+private:
+    Type val;
+};
+
+template<>
+void Wrapper<Point>::print() 
+{ 
+    std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl;
+}
+
+// template<typename T>    //#4
+// void variadicHelper(T&& first)  //#9
+// {
+//     Wrapper wrapper(std::forward<T>(first));    //#6
+//     wrapper.print();
+// }
+
+template<typename ... Args> 
+void variadicHelper();
+
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... args)
+{
+    Wrapper wrapper ( std::forward<T>(first) ); //#6
+    wrapper.print();
+    variadicHelper( std::forward<Args>(args) ...);  //#7
+}
+
+template<typename ... Args> 
+void variadicHelper(){}
```

Comment:
This doesn't need to be templated since you aren't using the template parameters inside the function.
- [ ] revise